excise/utils.py

```diff
@@ -0,0 +1,674 @@
+    OriginAddress2: Optional[str] = None
+
+    UserData: Optional[str] = None
+    CustomString1: Optional[str] = None
+    CustomString2: Optional[str] = None
+    CustomString3: Optional[str] = None
+    CustomNumeric1: Optional[Decimal] = None
+    CustomNumeric2: Optional[Decimal] = None
+    CustomNumeric3: Optional[Decimal] = None
+
+
+@dataclass
+class TransactionCreateRequestData:
+    EffectiveDate: str
+    InvoiceDate: str
+    TitleTransferCode: str
+    TransactionType: str
+    TransactionLines: List[TransactionLine]
+    InvoiceNumber: Optional[str] = None
+    Discount: Optional[Decimal] = Decimal("0.00")
+    UserTranId: Optional[str] = None
+
+
+def get_metadata_key(key_name: str):
+    """Namespace metadata key names: PLUGIN_ID:Key."""
+    return "mirumee.taxes.avalara_excise:" + key_name
+
+
+class EnhancedJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if dataclasses.is_dataclass(o):
+            return dataclasses.asdict(o)
+        if isinstance(o, Decimal):
+            return str(o)
+        return super().default(o)
+
+
+def get_api_url(use_sandbox=True) -> str:
+    """Based on settings return sanbox or production url."""
+    if use_sandbox:
+        return "https://excisesbx.avalara.com/api/v1/"
+    return "https://excise.avalara.com/api/v1/"
+
+
+def api_post_request(
+    url: str, data: Optional[Dict[str, Any]], config: AvataxConfiguration
+) -> Dict[str, Any]:
+    response = None
+    try:
+        auth = HTTPBasicAuth(
+            config.username_or_account,
+            config.password_or_license
+        )
+        headers = {
+            "x-company-id": config.company_name,
+            "Content-Type": "application/json",
+        }
+        formatted_data = json.dumps(data, cls=EnhancedJSONEncoder)
+        response = requests.post(
+            url,
+            headers=headers,
+            auth=auth,
+            data=formatted_data,
+        )
+        logger.debug("Hit to Avatax Excise to calculate taxes %s", url)
+        if response.status_code == 401:
+            logger.exception(
+                "Avatax Excise Authentication Error - Invalid Credentials"
+            )
+            return {}
+        json_response = response.json()
+        if json_response.get("Status") == "Errors found":
+            logger.exception(
+                "Avatax Excise response contains errors %s",
+                json_response
+            )
+            return json_response
+
+    except requests.exceptions.RequestException:
+        logger.exception("Fetching taxes failed %s", url)
+        return {}
+    except json.JSONDecodeError:
+        content = "Unable to find the response"
+        if response:
+            content = response.content
+        logger.exception(
+            "Unable to decode the response from Avatax Excise. "
+            "Response: %s", content
+        )
+        return {}
+    return json_response  # type: ignore
+
+
+def api_commit_transaction(
```

Comment:
Looks like a copy+paste logic from `api_post_request`. One difference which I see (if I am not wrong) is that you set data to `"{}"`. Why don't use `api_post_request`? Or create a wrapper that will trigger `api_post_request` inside of `api_commit_transaction` with data == {} ?

Reply:
Thanks! Added a fix here: https://github.com/saleor/saleor-plugin-avatax-excise/pull/7/commits/ce95dc0c8af772bf2e69d5262942480131d82c78